lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -274,6 +275,15 @@ public function getSchemaFromMetadata(array $classes)
 
             if (isset($class->table['uniqueConstraints'])) {
                 foreach ($class->table['uniqueConstraints'] as $indexName => $indexData) {
+                    $uniqIndex = new Index($indexName, $indexData['columns'], true, false, [], isset($indexData['options']) ? $indexData['options'] : []);
+
+                    foreach ($table->getIndexes() as $tableIndexName => $tableIndex) {
+                        if ($tableIndex->isFullfilledBy($uniqIndex)) {
+                            $table->dropIndex($tableIndexName);
+                            break;
```

Comment:
@Ocramius, yes, nothing denies adding more than one unique constraint for the same column in (DBAL) `Table` class (actually the commit that made it possible caused that issue), but that method creates table from metadata and adding a unique index for a table is possible when one of the conditions is met i.e.:
1. column has annotation `unique=true`;
2. column has relation `OneToOne`;
3. entity has annotation `uniqueConstraint`.

Therefore in modified method `getSchemaFromMetadata` table can have at most one fulfilled index in this loop, then I can't see any reason to continue search for another one.

I can remove it, but I'm worried that it could partially cover another bug, which should be resolved in another part. If you can provide an example to write one more unit test to prove that this `break` causes an issue I'll be glad and all will be clear when we have one more unit test for that case.